apps/mobile/src/app/wallet/wallet-settings/configure/[wallet]/[account]/choose-avatar.tsx

```diff
@@ -0,0 +1,165 @@
+        style={{ padding: 6 }}
+        borderRadius="round"
+        bg="ink.text-primary"
+      >
+        <Icon width={32} height={32} color={theme.colors['ink.background-primary']} />
+      </TouchableOpacity>
+    </Box>
+  );
+}
+
+function Avatars({
+  setNewIcon,
+  currentIcon,
+}: {
+  setNewIcon(icon: string): void;
+  currentIcon: string;
+}) {
+  return (
+    <Box flexDirection="row" gap="5" flexWrap="wrap">
+      {Object.values(iconNames).map(icon => {
+        return (
+          <AvatarButton
+            isSelected={currentIcon === icon}
+            onPress={() => {
+              setNewIcon(icon);
+            }}
+            key={icon}
+            icon={icon}
+          />
+        );
+      })}
+    </Box>
+  );
+}
+
+function ChooseAvatar({
+  fingerprint,
+  accountIndex,
+  account,
+}: {
+  fingerprint: string;
+  accountIndex: number;
+  account: Account;
+}) {
+  const { bottom } = useSafeAreaInsets();
+  const theme = useTheme<Theme>();
+  const dispatch = useAppDispatch();
+  const navigation = useNavigation();
+  const router = useRouter();
+  const [newIcon, setNewIcon] = useState<string | null>(null);
+  const { displayToast } = useToastContext();
+
+  useEffect(() => {
+    navigation.setOptions({ title: account.name });
+  }, [account.name, navigation]);
+
+  const isSubmitDisalbed = newIcon === null;
+
+  function setAvatar(icon: string) {
+    const payload = { fingerprint, accountIndex, icon };
+    dispatch(userUpdatesAccountIcon(payload));
+    router.back();
+    displayToast({ type: 'success', title: t`Account label updated` });
+  }
+
+  function onSubmit() {
+    if (newIcon) {
+      setAvatar(newIcon);
+    }
+  }
+
+  return (
+    <>
+      <Box flex={1} backgroundColor="ink.background-primary" justifyContent="space-between">
+        <ScrollView
+          contentContainerStyle={{
+            paddingTop: theme.spacing['5'],
+            paddingBottom: theme.spacing['5'] + bottom,
+            gap: theme.spacing[5],
+          }}
+        >
+          <Box px="5" gap="6">
+            <Text variant="heading03">{t`Choose account avatar`}</Text>
+            <Text variant="heading05">{t`Choose an image`}</Text>
+            <Avatars currentIcon={newIcon ?? account.icon} setNewIcon={setNewIcon} />
+            <Text variant="heading05">{t`Or choose one of your collectibles`}</Text>
+          </Box>
+        </ScrollView>
+        <Box
+          style={{ paddingBottom: bottom + theme.spacing['5'] }}
+          px="5"
+          pt="2"
+          bg="ink.background-primary"
+        >
+          <Button
+            disabled={isSubmitDisalbed}
+            onPress={onSubmit}
+            buttonState={isSubmitDisalbed ? 'disabled' : 'default'}
+            title={t`Confirm`}
+          />
+        </Box>
+      </Box>
+    </>
+  );
+}
+
+export default function ChooseAvatarScreen() {
+  const params = useLocalSearchParams();
+  if (!params.wallet || typeof params.wallet !== 'string') {
```

Comment:
```suggestion
  if (!params.wallet || isString(params.wallet)) {
```

there are type helpers

Reply:
I update this